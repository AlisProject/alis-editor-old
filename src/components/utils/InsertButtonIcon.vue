<template>
  <div class="editor-insert-button__icon" :class="{ 'is-active': active }">
    <div class="editor-insert-button__iconContent" v-html="src" />
  </div>
</template>

<script lang="ts">
export default {
  props: {
    active: Boolean,
    src: String
  }
}
</script>

<style>
.editor-insert-button__icon {
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  transition: all 0.15s ease-out;
}

.editor-insert-button__iconContent {
  transform: scale(0.6);
}

.editor-insert-button__icon .fill-target {
  fill: #5e5e5e;
}

.editor-insert-button__icon.is-active {
  background: #0086cc;
}

.editor-insert-button__icon.is-active .fill-target {
  fill: #fff;
}
</style>
