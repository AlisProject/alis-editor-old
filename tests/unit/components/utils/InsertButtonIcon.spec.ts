import { mount } from '@vue/test-utils'
import InsertButtonIcon from '@/components/utils/InsertButtonIcon.vue'

describe('InsertButtonIcon.vue', () => {
  test('renders correctly', () => {
    const wrapper = mount(InsertButtonIcon)
    expect(wrapper.element).toMatchSnapshot()
  })
})
