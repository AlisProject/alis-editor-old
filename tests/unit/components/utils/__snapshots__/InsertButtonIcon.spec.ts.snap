// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InsertButtonIcon.vue renders correctly 1`] = `
<div
  class="editor-insert-button__icon"
>
  <div
    class="editor-insert-button__iconContent"
  />
</div>
`;
